src/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -85,6 +86,14 @@ public static void IList_Contains()
             Assert.False(iList.Contains("1")); // Value not an int
         }
 
+
+        [Fact]
+        public static void AsReadOnly()
+        {
+            var array = new string[] { "a", "b" };
+            System.Collections.ObjectModel.ReadOnlyCollection<string> ro = Array.AsReadOnly(array);
```

Comment:
It'd be good to validate that this contains the same contents as the original / is the same as what you'd get from creating the collection manually, e.g.
```C#
Assert.Equal(new ReadOnlyCollection<string>(array), Array.AsReadOnly(array));
```